### PR TITLE
tests(ticdc): add upgrade gc safepoint integration tests

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Add an upgrade test that covers a case which changefeed checkpoint may be set to gc safepoint by old cdc. 

Issue Number:  ref https://github.com/pingcap/tiflow/issues/4115

### Check List 

Tests 

 - Integration test

Related changes

 - Need to cherry-pick to the release branch

### Release note 

```release-note
None
```

### CODE REVIEW

File: tests/integration_tests/upgrade_gc_safepoint/run.sh [hunk 1]
```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+
+set -e
```
Comment (lance6716) [inline tests/integration_tests/upgrade_gc_safepoint/run.sh:3]:
```suggestion
set -eu
```

I hope we can switch to `set -eu`, see https://github.com/pingcap/tiflow/pull/4160 .

also the problem of wrong position of `pd_cluster_id` in `gc_safepoint` can be found by it https://github.com/pingcap/tiflow/pull/4160#issuecomment-1002933474

Comment (overvenus) [reply tests/integration_tests/upgrade_gc_safepoint/run.sh]: Please go on with #4160, this PR will not be merged until #4115 is fixed.

**Pull request closed without merging**